R/tax_unique.R

```diff
@@ -0,0 +1,355 @@
+#' containing the class-level identifications.
+#' @param names \code{character}. The name of the column in the data frame
+#' containing the taxonomic names at mixed taxonomic levels; the data column
+#' "accepted_names" in a Paleobiology Database download is of this type.
+#' @param resolution \code{character}. The taxonomic resolution at which to
+#' identify unique occurrences, either species (the default) or genus.
+#'
+#' @return A \code{dataframe} of taxa, with each row corresponding to a unique
+#' "species" or "genus" in the dataset (depending on the chosen resolution).
+#' The dataframe will include the taxonomic information provided into the
+#' function, as well as a column providing the 'unique' names of each taxon.
+#'
+#' @details In many cases palaeobiologists count unique taxa by retaining only
+#' unique occurrences identified to the given taxonomic resolution, however
+#' here we also retain occurrences identified to a coarser resolution which are
+#' not already represented within the dataset.
+#' If we take the following set of occurrences:
+#' Albertosaurus sarcophagus
+#' Ankylosaurus sp.
+#' Aves indet.
+#' Ceratopsidae indet.
+#' Hadrosauridae indet.
+#' Ornithominus sp.
+#' Tyrannosaurus rex
+#' then a filter for species-level identifications will reduce the species
+#' diversity to two. However, none of these clades are nested within another, so
+#' each of the indeterminately identified occurrences represents at least one
+#' species not already represented in the dataset. This function is designed to
+#' deal with such taxonomic data, and would retain all seven 'species' in this
+#' dataset.
+#' Taxonomic information is supplied within a dataframe, in which columns
+#' provide identifications at different taxonomic levels. Occurrence
+#' data can be filtered to retain either unique species or unique genera. If a
+#' species-level filter is desired, the minimum input is for either 'binomial',
+#' 'species' and 'genus', or 'names' and 'genus' columns to be entered, as well
+#' as a 'family' column. In a standard Paleobiology Database dataframe, species
+#' information is only captured in the 'accepted_name' column, so a species-
+#' level filter should use 'genus = 'genus'' and 'names = 'accepted_name''
+#' arguments.If a genus-level filter is desired, the minimum input is for
+#' either 'binomial' or 'genus' columns to be entered, as well as a 'family'
+#' column.
+#' Missing data should be indicated with NAs, although the function can handle
+#' the e.g. "NO_FAMILY_SPECIFIED" labels within Paleobiology Database datasets.
+#' The function matches taxonomic names at increasingly higher taxonomic levels,
+#' so homonyms may be falsely filtered out.
+#'
+#' @section Developer(s):
+#' Bethany Allen
+#' @section Reviewer(s):
+#' Lewis A. Jones and William Gearty
+#'
+#' @examples
+#' #Retain unique species
+#' tax_unique(occdf = tetrapods, genus = "genus", family = "family", order =
```

Comment:
Yes, the rownames correspond to the original occdf rownames - as mentioned before, this is perhaps information that shouldn't be used at face value as we can arbitrarily filter out one equivalent occurrence over another. Just wanted to check that we're happy to erase this information!

Reply:
I don't think this information is useful for the output as it stands, so I'd remove it! Maybe something to think about for **future functionality** is having an option to change the output format. It would be nice to have a list or similar with each element containing occurrences for the 'unique taxon'.

Reply:
That's a nice idea! Perhaps another one to flag as an issue.